ipatests/test_integration/test_mkey_upgrade.py

```diff
@@ -0,0 +1,140 @@
+        cls.master.run_command(['systemctl', 'daemon-reload'])
+        tasks.install_master(cls.master, setup_dns=False)
+        tasks.install_replica(cls.master, cls.replicas[0], setup_dns=False)
+
+    def test_old_active_mkey(self):
+        p = re.compile('^KVNO: 1, Enctype: aes256-cts-hmac-sha1-96, .+ \\*$',
+                       flags=re.MULTILINE)
+
+        result = self.master.run_command(['kdb5_util', 'list_mkeys'])
+        assert p.search(result.stdout_text)
+        result = self.replicas[0].run_command(['kdb5_util', 'list_mkeys'])
+        assert p.search(result.stdout_text)
+
+    def test_enable_new_entypes(self):
+        base_dn = "dc=%s" % (",dc=".join(self.master.domain.name.split(".")))
+        realm = self.master.domain.name.upper()
+
+        entry_ldif = textwrap.dedent("""
+            dn: cn={realm},cn=kerberos,{base_dn}
+            changetype: modify
+            replace: krbSupportedEncSaltTypes
+            krbSupportedEncSaltTypes: aes128-sha2:normal
+            krbSupportedEncSaltTypes: aes128-sha2:special
+            krbSupportedEncSaltTypes: aes256-sha2:normal
+            krbSupportedEncSaltTypes: aes256-sha2:special
+            krbSupportedEncSaltTypes: aes256-cts:normal
+            krbSupportedEncSaltTypes: aes256-cts:special
+            krbSupportedEncSaltTypes: aes128-cts:normal
+            krbSupportedEncSaltTypes: aes128-cts:special
+            krbSupportedEncSaltTypes: camellia128-cts-cmac:normal
+            krbSupportedEncSaltTypes: camellia128-cts-cmac:special
+            krbSupportedEncSaltTypes: camellia256-cts-cmac:normal
+            krbSupportedEncSaltTypes: camellia256-cts-cmac:special
+            -
+            replace: krbDefaultEncSaltTypes
+            krbDefaultEncSaltTypes: aes256-sha2:special
+            krbDefaultEncSaltTypes: aes128-sha2:special
+            krbDefaultEncSaltTypes: aes256-cts:special
+            krbDefaultEncSaltTypes: aes128-cts:special""").format(
+            base_dn=base_dn, realm=realm)
+        tasks.ldapmodify_dm(self.master, entry_ldif)
+
+    def test_add_new_mkey(self):
+        self.master.run_command('kdb5_util add_mkey -e aes256-sha2 -s',
+                                stdin_text='Secret123\nSecret123')
+        time.sleep(2)
+
+    def test_new_inactive_mkey(self):
+        p = re.compile('^KVNO: 2, Enctype: aes256-cts-hmac-sha384-192, ',
+                       flags=re.MULTILINE)
+
+        result = self.master.run_command(['kdb5_util', 'list_mkeys'])
+        assert p.search(result.stdout_text)
+        result = self.replicas[0].run_command(['kdb5_util', 'list_mkeys'])
+        assert p.search(result.stdout_text)
+
+    def test_switch_mkey(self):
+        self.master.run_command(['kdb5_util', 'use_mkey', '2'])
+        time.sleep(2)
+
+    def test_new_active_mkey(self):
+        p = re.compile('^KVNO: 2, Enctype: aes256-cts-hmac-sha384-192, .+ \\*$',
+                       flags=re.MULTILINE)
```

Comment:
Should this also look for "Active"?

Reply:
The `*` at the end of the regexp means it is active.